### ISSUE TITLE
doc, x/build/dashboard: document Go 1.16 requires at least OpenBSD 6.4, drop OpenBSD 6.2 builders

### ISSUE THREAD

**Issue by dmitshur** [golang/go#42426]
Based on investigation and discussion in #42237 and #42408, and past support policy discussion in https://github.com/golang/go/issues/15227#issuecomment-293319876, we should drop OpenBSD 6.2 support from the upcoming Go 1.16 release. What's needed:

- [x] Disable `openbsd-{386,amd64}-62` builders for Go 1.16 and newer.
- [ ] Document in Go 1.16 release notes that Go 1.16 requires at least OpenBSD 6.4: "Go 1.15 was the last release that ran on OpenBSD 6.2. Go 1.16 requires OpenBSD 6.4 or newer."
   - Ideally we should say OpenBSD 6.7 or newer, but we don't have builders to back up such a claim (see #35712), so I think this is the best we can say for now.
   - Ideally we would've pre-announced this earlier, like in [Go 1.11 release notes](https://golang.org/doc/go1.11#ports). Unfortunately we didn't have the information we have now earlier.
- [ ] Update relevant [golang.org/wiki](https://golang.org/wiki) pages as needed.

Please comment if you think this plan can be improved.

CC @golang/release, @ianlancetaylor, @4a6f656c, @prattmic, @bcmills.

**Comment by dmitshur** [comment]
There seems to be agreement to move forward with this plan, so moving this issue to NeedsFix.

**Comment by dmitshur** [comment]
@4a6f656c It seems we're close to hopefully having the OpenBSD 6.8 builder working, given the update in https://github.com/golang/go/issues/42542#issuecomment-742594947.

I'm a bit concerned that stating in the release notes that OpenBSD 6.4 is supported is in conflict with the policy agreed on in #15227. According to that policy, Go 1.16 should support OpenBSD 6.8 and 6.7, and not support older OpenBSD releases.

It was difficult to find what to say when this issue was filed because we didn't have an OpenBSD 6.8 builder at all back then, and 6.4 was our latest (hence the "unfortunate" label). Now that it's changed (assuming we get the 6.8 builder working well soon, before RC 1), I wonder how much emphasis we should put on 6.4 still being supported.

**Comment by dmitshur** [comment]
We still need to resolve this by Go 1.16 final release, but to buy more time (e.g., to get #35712 completed), I don't think this needs to block beta 1, so I'll apply okay-after-beta1 label.

**Comment by dmitshur** [comment]
> It was difficult to find what to say when this issue was filed because we didn't have an OpenBSD 6.8 builder at all back then, and 6.4 was our latest (hence the "unfortunate" label). Now that it's changed (assuming we get the 6.8 builder working well soon, before RC 1), I wonder how much emphasis we should put on 6.4 still being supported.

Fortunately, we were able to get an OpenBSD 6.8 builder up and running (which uncovered some issues both in OpenBSD kernel and the Go project, but they've been resolved by now), and it is passing. Together with the OpenBSD 6.4 builder, we have more confidence that versions in between should be okay too.

The OpenBSD support policy covers the past two OpenBSD releases, which are 6.8 and 6.7 at this time. With the builders we now have, I don't think we need to say anything in the release notes about OpenBSD 6.2, since it's unsupported.

I updated https://golang.org/wiki/OpenBSD#building-from-source in wiki commit [`cb4b232b`](https://github.com/golang/go/wiki/OpenBSD/_compare/cb4b232b54791da32f85b1f11c03e59247632f14%5E...cb4b232b54791da32f85b1f11c03e59247632f14) to say that Go 1.15.x is the final Go version that ran on OpenBSD 6.2-6.3, since we know Go 1.16 will not work correctly on it.

Closing as this is resolved now.
